pkg/resourcemanager/controller/secret/reconciler.go

```
@@ -66,6 +67,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			// check if we are responsible for this MR, class might have changed, then we need to remove our finalizer
 			if ref.Name == secret.Name && r.ClassFilter.Responsible(&resource) {
 				secretIsReferenced = true
+				if _, ok := secret.Labels["managed-by"]; !ok {
+					secret.SetLabels(utils.MergeStringMaps(secret.GetLabels(), map[string]string{"managed-by": resource.Name}))
+					r.SourceClient.Update(ctx, secret)
```

The returned error is ignored.